### ISSUE TITLE
Pin rust version?

### ISSUE THREAD

**Issue by mccalluc** [opendp/opendp#1101]
In the process of checking rust source code formatting:
- #1027

I found that there are significant changes, even between 1.74.1 (which I had locally) and 1.74.0 (on CI).

Looking at `Cargo.toml`, an older version is specified, but that didn't seem to influence the behavior of `cargo fmt`:
```
[workspace.package]
...
rust-version = "1.64.0" ...
```

- What is `rust-version` in `Cargo.toml` supposed to do? Should I be surprised that it didn't determine the version used?
- Should we add a [toolchain file](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file) to the project? Or use `+` to specify version when running commands?
- What version of rust should we pin to?

**Comment by mccalluc** [comment]
- `rust-version` in `Cargo.toml` is just the minimum language version we would expect to work.
- The formatting changes aren't _really_ significant -- Michael provided more context: https://blog.rust-lang.org/2023/07/01/rustfmt-supports-let-else-statements.html
- Pinning the toolchain would give us a little more predictability, and should be easy, but nothing is burning.

I'll try to make a little PR.

**Comment by mccalluc** [comment]
I'm going to close this for a couple reasons:

The [docs](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file) don't push this as a default: _"Some projects find themselves 'pinned' to a specific release of Rust and want this information reflected in their source repository."_ This suggests to me that pinning is not the default, and instead Rust projects should expect to build in any version above the MSRV.

Second, if we did add a `rust-toolchain.toml`, it wouldn't be checked. This file has [lower precedence](https://rust-lang.github.io/rustup/overrides.html#overrides) than the `+` flag, which is set in `rust-build.sh` to `stable`:
```
TOOLCHAIN=stable
...
  [[ $TOOLCHAIN != UNSET ]] && CMD+=(+"$TOOLCHAIN")
```

If we did have problems with the rust build, the solution is probably to update the MSRV rather than to add a new constraint.
